example/multi-task/example_multi_task.py

```diff
@@ -0,0 +1,109 @@
+# pylint: skip-file
+import sys
+sys.path.insert(0, "../../python/")
+from data import mnist_iterator
+import mxnet as mx
+import numpy as np
+import logging
+import time
+
+logging.basicConfig(level=logging.DEBUG)
+
+def build_network():
+    data = mx.symbol.Variable('data')
+    fc1 = mx.symbol.FullyConnected(data = data, name='fc1', num_hidden=128)
+    act1 = mx.symbol.Activation(data = fc1, name='relu1', act_type="relu")
+    fc2 = mx.symbol.FullyConnected(data = act1, name = 'fc2', num_hidden = 64)
+    act2 = mx.symbol.Activation(data = fc2, name='relu2', act_type="relu")
+    fc3 = mx.symbol.FullyConnected(data = act2, name='fc3', num_hidden=10)
+    sm1 = mx.symbol.SoftmaxOutput(data = fc3, name = 'softmax1')
+    sm2 = mx.symbol.SoftmaxOutput(data = fc3, name = 'softmax2')
+
+    softmax = mx.symbol.Group([sm1, sm2])
+
+    return softmax
+
+class Multi_mnist_iterator(mx.io.DataIter):
+    '''multi label mnist iterator'''
+
+    def __init__(self, data_iter):
+        super(Multi_mnist_iterator, self).__init__()
+        self.data_iter = data_iter
+        self.batch_size = self.data_iter.batch_size
+
+    @property
+    def provide_data(self):
+        return self.data_iter.provide_data
+
+    @property
+    def provide_label(self):
+        provide_label = self.data_iter.provide_label[0]
+        # Different labels should be used here for actual application
+        return [('softmax1_label', provide_label[1]), \
+                ('softmax2_label', provide_label[1])]
```

Comment:
How to preprocess an actual multi-label dataset into proper format and write a data iterator for it?